matrix/Quaternion.hpp

```diff
@@ -280,9 +280,11 @@ class Quaternion : public Vector<Type, 4>
     void invert()
     {
         Quaternion &q = *this;
-        q(1) *= -1;
-        q(2) *= -1;
-        q(3) *= -1;
+        Type n = q.dot(q);
```

Comment:
`q.dot(q)` gives you the norm squared but you want to devide by the norm here.
I would just use the norm method that we already have here `n = q.norm()` which would use this correct implementation: https://github.com/PX4/Matrix/blob/master/matrix/Vector.hpp#L73